portal/models/reporting.py

```diff
@@ -119,49 +125,46 @@ def overdue_stats_by_org():
     return overdue_stats
 
 
-def generate_overdue_table_html(cutoff_days, overdue_stats, user, top_org):
-    cutoff_days.sort()
-
-    day_ranges = []
-    curr_min = 0
-    for cd in cutoff_days:
-        day_ranges.append("{}-{}".format(curr_min + 1, cd))
-        curr_min = cd
+def generate_and_send_summaries(cutoff_days, org_id):
+    ostats = overdue_stats_by_org()
+    cutoffs = [int(i) for i in cutoff_days.split(',')]
+    error_emails = set()
 
     ot = OrgTree()
-    rows = []
-    totals = defaultdict(int)
-
-    for org in sorted(overdue_stats, key=lambda x: x.name):
-        if top_org and not ot.at_or_below_ids(top_org.id, [org.id]):
-            continue
-        user_accessible = False
-        for user_org in user.organizations:
-            if ot.at_or_below_ids(user_org.id, [org.id]):
-                user_accessible = True
-                break
-        if not user_accessible:
-            continue
-        counts = overdue_stats[org]
-        org_row = [org.name]
-        curr_min = 0
-        row_total = 0
-        for cd in cutoff_days:
-            count = len([i for i in counts if ((i > curr_min) and (i <= cd))])
-            org_row.append(count)
-            totals[cd] += count
-            row_total += count
-            curr_min = cd
-        org_row.append(row_total)
-        rows.append(org_row)
-
-    totalrow = [_(u"TOTAL")]
-    row_total = 0
-    for cd in cutoff_days:
-        totalrow.append(totals[cd])
-        row_total += totals[cd]
-    totalrow.append(row_total)
-    rows.append(totalrow)
-
-    return render_template('site_overdue_table.html',
-                           ranges=day_ranges, rows=rows)
+    top_org = Organization.query.get(org_id)
+    if not top_org:
+        raise ValueError("No org with ID {} found.".format(org_id))
+    name_key = SiteSummaryEmail_ATMA.name_key(org=top_org.name)
+
+    for user in User.query.filter_by(deleted_id=None).all():
+        if (user.has_role(ROLE.STAFF) and user.email and (u'@' in user.email)
```

Comment:
I like to invert these conditions and `continue` if the necessary conditions aren't met, that way you don't have all of your code indented under an `if`, making it a little easier to read